File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int l = prices[0];
        int p = 0;
        for(int i = 1; i < prices.length; i++)
        {
            l = Math.min(l, prices[i]);
            p = Math.max(p, prices[i] - l);
        }
        return p;
    }
}

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.kt
class Solution {
    fun maxProfit(prices: IntArray): Int {
        var l = prices[0]
        var p = 0
        for(i in 1..prices.size - 1)
        {
            l = if (l < prices[i]) l else prices[i]
            p = if (p > prices[i] - l) p else prices[i] - l
        }
        return p;
    }
}